WebHostLib/upload.py

```diff
@@ -62,12 +62,19 @@ def upload_zip_to_db(zfile: zipfile.ZipFile, owner=None, meta={"race": False}, s
         elif file.filename.endswith(".archipelago"):
             try:
                 multidata = zfile.open(file).read()
-                MultiServer.Context._decompress(multidata)
             except:
                 flash("Could not load multidata. File may be corrupted or incompatible.")
                 multidata = None
 
     if multidata:
+        decompressed_multidata = MultiServer.Context._decompress(multidata)
+        leftover_names = [(name, decompressed_multidata["names"][0].index(name) + 1) for name in
+                          decompressed_multidata["names"][0] if name not in [slot.player_name for slot in slots]]
```

Comment:
Performance nitpick; `[slot.player_name for slot in slots]` is run and constructed for every iteration of the outer list comprehension, this'd be best created as a set beforehand to save a few cpu cycles.
Addditionally, instead of calling .index for every iteration (which scans the lists on each call), enumerate() should be used instead, which returns (index, element) pairs iterating once over the container.

Reply:
Changes made, check them out if you can. Thanks for the pointers.